pkg/operators/ebpf/formatters.go

```diff
@@ -143,10 +150,75 @@ func (i *ebpfInstance) initEnumFormatter(gadgetCtx operators.GadgetContext) erro
 	return nil
 }
 
+func (i *ebpfInstance) initStackConverter(gadgetCtx operators.GadgetContext) error {
+	var kernelSymbolResolver *kallsyms.KAllSyms = nil
+	for _, ds := range gadgetCtx.GetDataSources() {
+		for _, in := range ds.GetFieldsWithTag(StackIdKernelType) {
+			if in == nil {
+				continue
+			}
+			in.SetHidden(true, false)
+
+			if kernelSymbolResolver == nil {
+				var err error
+				kernelSymbolResolver, err = kallsyms.NewKAllSyms()
```

Comment:
I wonder if we can skip `kallsyms.NewKAllSyms()` if the `in` field is already hidden.

If the `print-stack` ebpfParam is false, then we don't need to spend CPU time parsing kallsyms.

But that would be an optimization for another PR.